power_grid_model_c/power_grid_model/include/power_grid_model/main_core/input.hpp

```diff
@@ -158,6 +163,13 @@ inline void add_component(MainModelState<ComponentContainer>& state, ForwardIter
             emplace_component<Component>(state, id, input, regulated_object_type, u_rated);
         }
     }
+    // Make sure that each regulated object has at most one regulator
+    const std::unordered_set<Idx2D, Idx2DHash> unique_regulated_objects(regulated_objects.begin(),
+                                                                        regulated_objects.end());
+    if (unique_regulated_objects.size() != regulated_objects.size()) {
+        // There are duplicates
+        throw DuplicativelyRegulatedObject{};
```

Comment:
i guess we'll need the input validation checks to get the actual duplicate ids for efficiency

Reply:
I'd assume the same